### PR TITLE
Benchmark: enable transform_path

### PR DESCRIPTION
**Pull request by wbo4958**
If we directly use train dataframe as the evaluation dataframe, the RMSE or accuracy may suffer the overfitting issue. This PR enables transform_path parameter which the user can specify, if it is not set, the train dataframe will be as the eval dataframe.

### CODE REVIEW

File: benchmark/benchmark/base.py [hunk 1]
```diff
@@ -154,13 +154,13 @@ def args(self) -> argparse.Namespace:
     def class_params(self) -> Dict[str, Any]:
         return self._class_params
 
-    def train_df(
-        self, spark: SparkSession
+    def input_dataframe(
+        self, spark: SparkSession, *paths: str
```
Comment (leewyang) [inline benchmark/benchmark/base.py:158]: Should probably rename the `train_df` variable on L197-L198 as well.

**Review by leewyang — COMMENTED** [review]
Just minor variable naming.

**Review by eordentlich — COMMENTED** [review]
would be good to add support for creating training and test files via train_test_split in gen_data.

One alternative to having a separate test file is to use: pyspark.sql.DataFrame.randomSplit to create the train and test portions, with a fixed seed so that spark rapids ml and spark ml operate on the same files.

File: benchmark/benchmark/base.py (hunk 1 above)
Comment (wbo4958) [reply benchmark/benchmark/base.py:158]: Done. Thx

File: benchmark/benchmark/base.py [hunk 2]
```diff
@@ -238,6 +249,7 @@ def run_once(
         spark: SparkSession,
         df: DataFrame,
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply benchmark/benchmark/base.py]: Done. Thx

File: benchmark/benchmark/bench_kmeans.py [hunk 3]
```diff
@@ -117,6 +117,7 @@ def run_once(
         spark: SparkSession,
         df: DataFrame,
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply benchmark/benchmark/bench_kmeans.py]: Done. Thx

File: benchmark/benchmark/bench_pca.py [hunk 4]
```diff
@@ -114,6 +114,7 @@ def run_once(
         spark: SparkSession,
         df: DataFrame,
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply benchmark/benchmark/bench_pca.py]: Done. Thx

**Review by leewyang — APPROVED** [review]

**Review by leewyang — APPROVED** [review]
